invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/UserDetails/components/ItemsSearch/ItemsSearch.js

```diff
@@ -37,29 +38,31 @@ export default class ItemsSearch extends Component {
   };
 
   executeSearch = queryString => {
-    if (queryString) {
-      return this.fetchItems(queryString);
-    } else {
-      return this.fetchItems(this.props.queryString);
-    }
+    queryString = queryString || this.props.queryString;
+    this.setState({ prevSearchQuery: queryString });
+
+    return this.fetchItems(queryString);
   };
 
-  _onPasteHandler = e => {
+  _onPasteHandler = async e => {
     let queryString = e.clipboardData.getData('Text');
-    this.executeSearch(queryString).then(data => {
-      if (this.props.items.hits[0].status === 'LOANABLE') {
-        this.checkoutItem(this.props.items.hits[0], this.props.patron).then(
-          () => {
-            this.clearResults();
-            this.fetchUpdatedCurrentLoans(this.props.patron);
-          }
-        );
+
+    if (queryString) {
+      await this.executeSearch(queryString);
+
+      const { hits } = this.props.items;
+      const hasOneHit = !_isEmpty(hits) && hits.length === 1;
+      if (hasOneHit && hits[0].metadata.status === 'LOANABLE') {
```

Comment:
warning, it will break because now it is `CAN_CIRCULATE`

Reply:
Thanks!